qt/applications/workbench/workbench/plotting/figuremanager.py

```diff
@@ -96,14 +95,14 @@ def deleteHandle(self, _, workspace):
                 to_redraw = ax.remove_workspace_artists(workspace)
             else:
                 to_redraw = False
-            # We check for axes type below as a pseudo check for an axes being
+            # We check for images and lines below as a pseudo check for an axes being
             # a colorbar. Creating a colorfill plot creates 2 axes: one linked
             # to a workspace, the other a colorbar. Deleting the workspace
             # deletes the colorfill, but the plot remains open due to the
             # non-empty colorbar. This solution seems to work for the majority
             # of cases but could lead to unmanaged figures only containing an
             # Axes object being closed.
-            if type(ax) is not Axes:
+            if len(ax.images) != 0 or len(ax.lines) != 0:
```

Comment:
Quick question. I think we are starting to put this type of check in several places, i.e does this `Axes` instance only contain a colorbar? Should this go into a function in `mantid.plots.utility` to avoid duplicating it?